OpenEdXMobile/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -152,6 +159,34 @@ private void init() {
         }
     }
 
+    public void showBanner(LoginAPI loginAPI, boolean delayCall) {
+        if (delayCall) {
+            new Handler().postDelayed(() -> callBannerAPI(loginAPI), 10000);
+        } else {
+            callBannerAPI(loginAPI);
+        }
+    }
+
+    private void callBannerAPI(LoginAPI loginAPI) {
+        loginAPI.getUnacknowledgedNotice().enqueue(new Callback<UnacknowledgedNoticeResponse>() {
+            @Override
+            public void onResponse(Call<UnacknowledgedNoticeResponse> call, Response<UnacknowledgedNoticeResponse> response) {
+                if (response.isSuccessful() && response.code() == HttpStatus.OK) {
```

Comment:
add check `environment.loginPrefs.username != null` otherwise, register and sig-in footer will visible with the banner.